echoregions/lines/lines.py

```diff
@@ -212,42 +250,119 @@ def filter_bottom(bottom, start_date, end_date):
+                    }
+                )
+
+                # Concat new corner rows to bottom points
+                bottom_points = pd.concat(
+                    [left_side_new_rows, bottom_points, right_side_new_rows]
+                ).reset_index(drop=True)
+
+                # Convert region time to integer timestamp.
+                bottom_points_with_int_timestamp = bottom_points.copy()
+                bottom_points_with_int_timestamp["time"] = matplotlib.dates.date2num(
+                    bottom_points_with_int_timestamp["time"]
+                )
+
+                # Convert ping_time to unix_time since the masking does not work on datetime objects.
+                da_Sv = da_Sv.assign_coords(
+                    unix_time=(
+                        "ping_time",
+                        matplotlib.dates.date2num(da_Sv.coords["ping_time"].values),
+                    )
+                )
+
+                # Create bottom mask
+                # bottom: True, otherwise: False
+                r = regionmask.Regions(
+                    outlines=[np.array(bottom_points_with_int_timestamp)],
+                    overlap=False,
+                )
+                bottom_mask = xr.where(
+                    np.isnan(
+                        r.mask(
+                            da_Sv["unix_time"],
+                            da_Sv["depth"],
+                            wrap_lon=False,
+                        )
+                    ),
+                    False,
+                    True,
+                )
+            else:
+                # create joint index
+                joint_index = list(
+                    set(list(pd.DataFrame(echogram_ping_time)[0]) + list(filtered_bottom.index))
+                )
+
+                # Interpolate on the echogram coordinates. Note that some interpolation kwaargs
+                # will result in some interpolation NaN values. The ffill and bfill lines
+                # are there to fill in these NaN values.
+                # TODO There exists a problem where when we use .loc prior to reindexing
+                # we are hit with a key not found error.
+                bottom_points = (
+                    filtered_bottom[["depth"]]
+                    .reindex(joint_index)
+                    .interpolate(**kwargs)
+                    .loc[echogram_ping_time]
+                    .ffill()
+                    .bfill()
+                )
+
+                # convert to data array for bottom
+                bottom_da = bottom_points["depth"].to_xarray()
+                bottom_da = bottom_da.rename({"time": "ping_time"})
+
+                # create a data array of depths
+                depth_da = da_Sv["depth"] + xr.zeros_like(da_Sv)
+
+                # create a mask for the bottom:
+                # bottom: True, otherwise: False
+                bottom_mask = depth_da >= bottom_da
```

Comment:
The regionmask returns 1's and 0's, while the below/above returns True/Falses. Is my reading correct? If so, could you synchronize it, and synchronize with the echoregions' regions masking funcitonality.

Reply:
At the very end, we actually turn booleans to 1s/0s:
```
        # Bottom: True becomes 1, False becomes 0
        bottom_mask = bottom_mask.where(True, 1, 0)
```
Also, regionmask returns NaNs and non-NaNs since I'm using regionmask's 2D option here.